### PR TITLE
Fix mypy complaints

### PR DESCRIPTION
**Pull request by Lash-L**
With py.typed enabled there are a lot of mypy complaints on the component side using the library.

This makes a few changes such as making some variables that we require/ know we will have into empty versions of themselves before we get them, i.e. Status.

Exporting some commonly accessed values like the name of the dock error

In the old system it was 

props.status.dock_error.name But status could be none and dock_error could be None and that was two additional None checks that had to be done in the component side - this simplifies things.

### CODE REVIEW

**Comment by humbertogontijo** [comment]
I'm ok with the new field within Status class

Something else is wrong with typing. RoborockBase.from_dict can return None

**Comment by Lash-L** [comment]
> I'm ok with the new fields within Status class
> 
> 
> 
> Something else is wrong with typing. RoborockBase.from_dict can return None

Hmm

Should it raise when from dict is passed a non dict and returns none? Or create an empty version of the class.

I see what you mean now. What's you opinion on best typing there?

**Comment by humbertogontijo** [comment]
Raising an exception is a good way to type it. But at the same time it could have a big impact on core and hacs implementation

**Comment by Lash-L** [comment]
Hows that? A few guaranteed types - others optional None

File: roborock/api.py [hunk 1]
```diff
@@ -413,9 +410,12 @@ async def send_command(
         return response
 
     async def get_status(self) -> Status:
-        return self._status_type.from_dict(await self.cache[CacheableAttribute.status].async_value())
+        data = self._status_type.from_dict(await self.cache[CacheableAttribute.status].async_value())
+        if data is None:
+            return Status()
```
Comment (humbertogontijo) [inline roborock/api.py]: self._status_type() maybe?

File: roborock/api.py [hunk 2]
```diff
@@ -293,10 +293,7 @@ def on_message_received(self, messages: list[RoborockMessage]) -> None:
                     payload = data.payload[0:24]
                     [endpoint, _, request_id, _] = struct.unpack("<8s8sH6s", payload)
                     if endpoint.decode().startswith(self._endpoint):
-                        try:
-                            decrypted = Utils.decrypt_cbc(data.payload[24:], self._nonce)
-                        except ValueError as err:
-                            raise RoborockException("Failed to decode %s for %s", data.payload, data.protocol) from err
+                        decrypted = Utils.decrypt_cbc(data.payload[24:], self._nonce)
```
Comment (humbertogontijo) [inline roborock/api.py]: Why raise ValueError?

Comment (Lash-L) [reply roborock/api.py]: That's what the underlying package raises

Comment (Lash-L) [reply roborock/api.py]: Although 0 i'm not sure why i deleted it. thta was a mistake i will add back

File: roborock/api.py (hunk 1 above)
Comment (Lash-L) [reply roborock/api.py]: good call

**Comment by Lash-L** [comment]
I'll fix lint commit in merge

**Review by humbertogontijo — APPROVED** [review]

**Pull request merged**